android/src/main/java/com/gaspardbruno/staticsafeareainsets/RNStaticSafeAreaInsetsModule.java

```diff
@@ -28,23 +35,59 @@ public String getName() {
   @Override
   public Map<String, Object> getConstants() {
     final Map<String, Object> constants = new HashMap<>();
-
-    if (getCurrentActivity() != null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-      final Activity activity = getCurrentActivity();
+    // Window insets are parts of the window that are covered by system views (status bar,
+    // navigation bar, notches). There are no apis the get these values for android < M so we
+    // do a best effort polyfill.
+    EdgeInsets windowInsets = new EdgeInsets(0, 0, 0, 0);
+    final Activity activity = getCurrentActivity();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
       final View view = activity.getWindow().getDecorView();
       final WindowInsets insets = view.getRootWindowInsets();
+      windowInsets = new EdgeInsets(
+          insets.getSystemWindowInsetTop(),
+          insets.getSystemWindowInsetRight(),
+          insets.getSystemWindowInsetBottom(),
+          insets.getSystemWindowInsetLeft());
+    }
+    final Resources resources = mReactContext.getResources();
```

Comment:
Hello @Rapsssito Could you please explain what these next lines (52 - 63) do? We are having trouble understanding why you are using `Resources` and all this logic with `PixelUtil`

Reply:
`Resources` are used for devices below `VERSION_CODES.M` that do not have `WindowInsets` available. Using `Resources` we get the `"status_bar_height"` and `"navigation_bar_height"` to calculate the insets manually.

Reply:
`PixelUtil` is used for getting the DIP, instead of the PX, while addressing the inset dimensions. React Native works with DIP.